spec/services/hmrc/interface/result_service_spec.rb

```diff
@@ -0,0 +1,163 @@
+        { 'income/sa/pensions_and_state_benefits/selfAssessment': [] },
+        { 'income/sa/source/selfAssessment': [] },
+        { 'income/sa/employments/selfAssessment': [] },
+        { 'income/sa/additional_information/selfAssessment': [] },
+        { 'income/sa/partnerships/selfAssessment': [] },
+        { 'income/sa/uk_properties/selfAssessment': [] },
+        { 'income/sa/foreign/selfAssessment': [] },
+        { 'income/sa/further_details/selfAssessment': [] },
+        { 'income/sa/interests_and_dividends/selfAssessment': [] },
+        { 'income/sa/other/selfAssessment': [] },
+        { 'income/sa/summary/selfAssessment': [] },
+        { 'income/sa/trusts/selfAssessment': [] },
+        { 'employments/paye/employments': [] },
+        { 'benefits_and_credits/working_tax_credit/applications': [] },
+        { 'benefits_and_credits/child_tax_credit/applications': [] }
+      ]
+    }
+  end
+  let(:expected_status) { 200 }
+  before do
+    stub_request(:post, %r{http.*laa-hmrc-interface.*apps.live-1.cloud-platform.service.justice.gov.uk/oauth/token})
+      .to_return(
+        status: 200,
+        body: '{"access_token":"test-bearer-token","token_type":"Bearer","expires_in":7200,"created_at":1582809000}',
+        headers: { 'Content-Type' => 'application/json; charset=utf-8' }
+      )
+    stub_request(:get, %r{http.*laa-hmrc-interface.*apps.live-1.cloud-platform.service.justice.gov.uk/api/v1/submission/result/#{hmrc_response.submission_id}})
+      .to_return(
+        status: expected_status,
+        body: expected_body.to_json,
+        headers: { 'Content-Type' => 'application/json; charset=utf-8' }
+      )
+  end
+
+  describe '#call' do
+    subject(:call) { described_class.call(hmrc_response) }
+
+    it { is_expected.to eql expected_body }
+  end
+
+  describe '.call' do
+    subject(:call) { submission_result.call }
+
+    context 'when the submission has completed successfully' do
+      it { is_expected.to eql expected_body }
+    end
+
+    context 'when the submission is still in progress' do
+      let(:expected_status) { 202 }
+      let(:expected_body) do
+        {
+          submission: hmrc_response.submission_id,
+          status: 'in_progress',
+          _links: [
+            {
+              href: "https://main-laa-hmrc-interface-uat.apps.live-1.cloud-platform.service.justice.gov.uk/api/v1/submission/status/#{hmrc_response.submission_id}"
+            }
+          ]
+        }
+      end
+
+      it 'does not raise an error' do
+        expect { call }.not_to raise_error
+      end
+    end
+
+    context 'when the submission has failed' do
+      let(:expected_status) { 202 } # TODO: Response should be 200, also json response should equal - AP-2639 addresses
+      # {
+      #   data: [
+      #     { correlation_id: hmrc_response.submission_id },
+      #     { error: 'submitted client details could not be found in HMRC service' }
+      #   ]
+      # }
```

Comment:
Can this be removed?

Reply:
It was kind of the continuation of the TODO ...also json response should equal
For when the updates get made to the interface

Reply:
Sorry, didn't spot the comment on the previous line 🤦 That makes sense to me.